### PR TITLE
Call GovPay from API and redirect User

### PR DESCRIPTION
**Pull request by jphillips2121**
- Refactored GetPaymentSession to GetPayment and added to own payment service folder
- Added PatchPayment which calls API to payment service folder
- Created externalpayment service folder and added createExternalPayment call to API
- Added Patch and createExternal to controller to call GovPay URL and redirect user 
- Added Unit Tests for all services

Resolves: CPS-156

### Type of change

* [X] New feature

### Pull request checklist

* [X] I have added unit tests for new code that I have added
* [X] I have added/updated functional tests where appropriate
* [X] The code follows our [coding standards](https://github.com/companieshouse/styleguides/blob/master/go.md)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -26,11 +26,10 @@
         <junit-jupiter-engine.version>5.2.0</junit-jupiter-engine.version>
         <mockito-junit-jupiter.version>2.18.0</mockito-junit-jupiter.version>
         <structured-logging.version>1.5.0-rc2</structured-logging.version>
-        <java-session-handler.version>2.0.0-rc4</java-session-handler.version>
+        <java-session-handler.version>2.1.0-rc2</java-session-handler.version>
```
Comment (divya051988) [inline pom.xml]: Would you want to use the latest release 2.2.0 ?

Comment (jphillips2121) [reply pom.xml]: Updated

File: src/main/java/uk/gov/companieshouse/web/payments/service/externalpayment/ExternalPaymentService.java [hunk 2]
```diff
@@ -0,0 +1,9 @@
+package uk.gov.companieshouse.web.payments.service.externalpayment;
+
+import uk.gov.companieshouse.web.payments.exception.ServiceException;
+
```
[the remark this answers is not in the document]
Comment (jphillips2121) [reply src/main/java/uk/gov/companieshouse/web/payments/service/externalpayment/ExternalPaymentService.java]: Added

File: src/main/java/uk/gov/companieshouse/web/payments/service/payment/PaymentService.java [hunk 3]
```diff
@@ -1,10 +1,13 @@
-package uk.gov.companieshouse.web.payments.service;
+package uk.gov.companieshouse.web.payments.service.payment;
 
 import uk.gov.companieshouse.web.payments.exception.ServiceException;
 import uk.gov.companieshouse.web.payments.model.PaymentSummary;
 
 public interface PaymentService {
```
[the remark this answers is not in the document]
Comment (jphillips2121) [reply src/main/java/uk/gov/companieshouse/web/payments/service/payment/PaymentService.java]: Added

**Pull request merged**